src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -506,6 +509,10 @@ extern "C" void PalAttachThread(void* thread)
 extern "C" bool PalDetachThread(void* thread)
 {
     UNREFERENCED_PARAMETER(thread);
+    if (g_threadDetachCallback != nullptr)
```

Comment:
Seeing the options, I think we can go with the callback ... but the callback registration should go directly from CoreLib to MRT. For example, like RhRegisterGcCallout is setup today.